integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -207,3 +212,34 @@ async def resync_pipelines(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                         {**pipeline.asdict(), "__project": project.asdict()}
                         for pipeline in pipelines_batch
                     ]
+
+
+@ocean.on_resync(ObjectKind.MEMBER)
+async def resync_members(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    gitlab_resource_config: GitlabMembersResourceConfig = typing.cast(
+        GitlabMembersResourceConfig, event.resource_config
+    )
+    selector = gitlab_resource_config.selector
+
+    async def process_group_members(service, group):
+        members = [
+            member
+            async for members_batch in service.get_all_group_members(group)
+            for member in members_batch
+        ]
+
+        if selector.enrich_with_public_email:
+            enriched_member_tasks = [
+                service.enrich_member_with_public_email(member) for member in members
+            ]
+            enriched_members = await asyncio.gather(*enriched_member_tasks)
+            return enriched_members
+
+        return [member.asdict() for member in members]
+
+    for service in get_cached_all_services():
+        async for groups in service.get_all_groups(skip_validation=True):
+            group_tasks = [process_group_members(service, group) for group in groups]
+            for group_task in asyncio.as_completed(group_tasks):
+                group_members = await group_task
+                yield group_members
```

Comment:
what will happen when a member is in multiple groups? will it get overwritten in port with the latest group that was fetched?

Reply:
Yes it would, a function to cache members and return only unsynced / uncached members can solve this. So we don't have to hit port with members that have been synced already